extensions/sql-migration/src/api/sqlUtils.ts

```diff
@@ -127,6 +127,32 @@ export interface LoginTableInfo {
 	status: string;
 }
 
+export async function getSourceConnectionProfile(): Promise<azdata.connection.ConnectionProfile> {
```

Comment:
Majority of the changes in this PR revolve around refactoring these commonly used helpers, and replacing references to possibly-stale connection info with these calls to always fetch the latest connection info